hibernate-core/src/main/java/org/hibernate/boot/archive/internal/StandardArchiveDescriptorFactory.java

```diff
@@ -130,12 +130,14 @@ public URL adjustJarFileEntryUrl(URL url, URL rootUrl) {
 			}
 			catch (MalformedURLException e) {
 				// allow to pass through to return the original URL
-				log.debugf(
-						e,
-						"Unable to adjust relative <jar-file/> URL [%s] relative to root URL [%s]",
-						filePart,
-						rootUrlFile.getAbsolutePath()
-				);
+				if ( log.isDebugEnabled() ) {
+					log.debugf(
+							e,
+							"Unable to adjust relative <jar-file/> URL [%s] relative to root URL [%s]",
+							filePart,
+							rootUrlFile.getAbsolutePath()
```

Comment:
I added logging level checking whenever the method invoking is involved. We never know how much performance cost will be paid for future coding might provide an overridden heavy weight method implementation. Adding logging level checking is simple and future proof.

Reply:
Yes, it *is* future proof.  For whatever reason, and I may be in the minority which is fine, I prefer to not do the level check in trivial cases.  I don't think it is as easy as saying "well anything that calls a method.  E.g. here calling `#getAbsolutePath()` does have an overhead, so this is fine.  The reason I prefer to avoid the check is ofc because the logging system does the same checks already.  I'm just talking in the general case as a reply to your comment.  Again, here it is fine

Reply:
Yeah, this is not clear-cut decision and it is hard to decide which scenario is trivial. Yeah, method invoking might not be a good criterion.